File: js/advanced/6a_find_middle_node.js
//Find the middle node (if size = even, return 2nd middle node).

var middleNode = function(head) {
    let slow = head
    let fast = head
    while(fast && fast.next){
        slow = slow.next
        fast = fast.next
        if(fast) fast = fast.next
    }
    return slow
};

File: c++/advanced/6a_find_middle_node.cpp
//Find the middle node (if size = even, return 2nd middle node).

ListNode* middleNode(ListNode* head) {
        ListNode* slow = head;
        ListNode* fast = head;
        while(fast->next){
            slow = slow->next;
            fast = fast->next;
            if(fast->next) fast = fast->next;
        }
        return slow;
    }